pytest_fixtures/component/provision_pxe.py

```diff
@@ -189,39 +191,41 @@ def module_provisioning_sat(
     # we might need to set up Sat's DNS server as the primary one on the Sat host
     provisioning_upstream_dns_primary = (
         broker_data_out.provisioning_upstream_dns
-        if settings.server.is_ipv6
+        if sat_ipv6
         else broker_data_out.provisioning_upstream_dns.pop()
     )  # There should always be at least one upstream DNS
     provisioning_upstream_dns_secondary = (
         broker_data_out.provisioning_upstream_dns.pop()
-        if len(broker_data_out.provisioning_upstream_dns) and not settings.server.is_ipv6
+        if len(broker_data_out.provisioning_upstream_dns) and not sat_ipv6
         else None
     )
 
     domain = sat.api.Domain(
         location=[module_location],
         organization=[module_sca_manifest_org],
-        dns=None if settings.server.is_ipv6 else module_provisioning_capsule.id,
+        dns=None if sat_ipv6 else module_provisioning_capsule.id,
         name=provisioning_domain_name,
     ).create()
 
     subnet = sat.api.Subnet(
         location=[module_location],
         organization=[module_sca_manifest_org],
         network=str(provisioning_network.network_address),
-        network_type='IPv6' if settings.server.is_ipv6 else 'IPv4',
+        network_type=NetworkType.IPV4.formatted
+        if sat.network_type == NetworkType.IPV4
+        else NetworkType.IPV6.formatted,
         mask=str(provisioning_network.netmask),
         gateway=broker_data_out.provisioning_gw_ip,
         from_=broker_data_out.provisioning_host_range_start,
         to=broker_data_out.provisioning_host_range_end,
         dns_primary=provisioning_upstream_dns_primary,
         dns_secondary=provisioning_upstream_dns_secondary,
         boot_mode='DHCP',
-        ipam='None' if settings.server.is_ipv6 else 'DHCP',
-        dhcp=None if settings.server.is_ipv6 else module_provisioning_capsule.id,
+        ipam='None' if sat_ipv6 else 'DHCP',
+        dhcp=None if sat_ipv6 else module_provisioning_capsule.id,
         tftp=module_provisioning_capsule.id,
         template=module_provisioning_capsule.id,
-        dns=None if settings.server.is_ipv6 else module_provisioning_capsule.id,
+        dns=None if sat_ipv6 else module_provisioning_capsule.id,
```

Comment:
And here. Related discussion if this is valid to begin with in https://github.com/theforeman/foreman-documentation/pull/3645.

Reply:
Same https://github.com/SatelliteQE/robottelo/pull/17904#discussion_r2068469195